dvc/path_info.py

```diff
@@ -19,6 +19,7 @@
 
 
 class PathInfo(pathlib.PurePath):
+    __slots__ = ()
```

Comment:
Could you elaborate on what is going on here and down below, please?

Reply:
`pathlib.PurePath` uses [slots](https://docs.python.org/3/reference/datamodel.html#slots) to make itself faster, these prevent a creation of `__dict__`, make objects smaller and speed up attribute access. When you subclass such object you need to specify `__slots__` again to force this optimization on a new class. Empty `__slots__` mean we don't add new attributes. 

`_cparts` is a list of casefolded parts of a path. `pathlib` uses this property to compare `Path` objects (`__eq__`, `__gt__`, `__lt__` and friends). We use them to compare start of `self` with `other`.

Reply:
Thank you! Could we add these as a comment? Would be a shame to lose it in PR.

Reply:
Slots is built-in python mechanic, do you really want me to comment on it?
And there is a comment about `_cparts` below.

Reply:
This is not at all obvious, so let's add a comment about it, so anyone who is not Python-fu master could understand it too, or at least have some hints :slightly_smiling_face: